app/assets/javascripts/discourse/app/components/glimmer-header/icons.gjs

```diff
@@ -23,12 +23,21 @@ export default class Icons extends Component {
   @service header;
   @service search;
 
+  _isStringType = (icon) => typeof icon === "string";
+
   <template>
     <ul class="icons d-header-icons">
-      {{#each _extraHeaderIcons as |icon|}}
-        <MountWidget @widget={{icon}} />
-        {{! I am not sure how we are going to render glimmer components here without
-        being able to import them. }}
+      {{#each _extraHeaderIcons as |Icon|}}
+        {{#if (this._isStringType Icon)}}
+          <MountWidget @widget={{Icon}} />
```

Comment:
IIRC this doesn't work reliably, because widget icons tend to rely on `parentWidget.state` and stuff?

So... should we just drop this and say 'widget-based icons will not render on the glimmer header'?

Reply:
We need to keep this in place, for example the chat widget is a rendered as a string. We don't pass around state, I think it is important we leave it as is for now.